plugin/pkg/scheduler/algorithm/scheduler_interface.go

```diff
@@ -26,8 +26,9 @@ import (
 // managed by Kubernetes.
 type SchedulerExtender interface {
 	// Filter based on extender-implemented predicate functions. The filtered list is
-	// expected to be a subset of the supplied list.
-	Filter(pod *api.Pod, nodes *api.NodeList) (filteredNodes *api.NodeList, err error)
+	// expected to be a subset of the supplied list. The returned failedNodesMap are added to
+	// list the filtered out nodes information.
```

Comment:
Would like to modify the comment to:

"failedNodesMap optionally contains the list of failed nodes and failure reasons."

This covers the upgrade cases mentioned earlier.